scripts/functest.sh

```diff
@@ -0,0 +1,122 @@
+${CSCLI} decisions list -r 4.4.0.0/16 -o json --contained | ${JQ} '.[].decisions[0].value == "4.4.4.0/24"'  || fail
+#check subrange
+${CSCLI} decisions list -r 4.4.4.2/28 -o json | ${JQ} '.[].decisions[0].value == "4.4.4.0/24"'  || fail
+${CSCLI} decisions list -r 4.4.3.2/28 -o json | ${JQ} '. == null'  || fail
+
+###
+### ipv6 tests
+###
+init
+APIK=`${CSCLI} bouncers add TestingBouncer -o=raw`
+
+#
+# ip ipv6 tests
+#
+#add ip decision
+${CSCLI} decisions add -i 1111:2222:3333:4444:5555:6666:7777:8888
+${CSCLI} decisions list -o json | jq -e '.[].decisions[0].value == "1111:2222:3333:4444:5555:6666:7777:8888"'  || fail
+#check ip matching/unmatching
+${CSCLI} decisions list -i 1111:2222:3333:4444:5555:6666:7777:8888 -o json | jq -e '.[].decisions[0].value == "1111:2222:3333:4444:5555:6666:7777:8888"'  || fail
+${CSCLI} decisions list -i 1211:2222:3333:4444:5555:6666:7777:8888 -o json | jq -e '. == null' || fail
+${CSCLI} decisions list -i 1111:2222:3333:4444:5555:6666:7777:8887 -o json | jq -e '. == null' || fail
+#check outer range
+${CSCLI} decisions list -r 1111:2222:3333:4444:5555:6666:7777:8888/48 -o json | jq -e '. == null' || fail
+${CSCLI} decisions list -r 1111:2222:3333:4444:5555:6666:7777:8888/48 --contained -o json | jq -e '.[].decisions[0].value == "1111:2222:3333:4444:5555:6666:7777:8888"' || fail
+${CSCLI} decisions list -r 1111:2222:3333:4444:5555:6666:7777:8888/64 -o json | jq -e '. == null' || fail
+${CSCLI} decisions list -r 1111:2222:3333:4444:5555:6666:7777:8888/64 -o json --contained | jq -e '.[].decisions[0].value == "1111:2222:3333:4444:5555:6666:7777:8888"'  || fail
+#
+# range ipv6 test
+#
+#add range decision
+${CSCLI} decisions add -r aaaa:2222:3333:4444::/64
+${CSCLI} decisions list -o json | jq -e '.[0].decisions[0].value == "aaaa:2222:3333:4444::/64", .[1].decisions[0].value == "1111:2222:3333:4444:5555:6666:7777:8888"' || fail
+#check ip within/out of range
+${CSCLI} decisions list -i aaaa:2222:3333:4444:5555:6666:7777:8888 -o json | jq -e '.[].decisions[0].value == "aaaa:2222:3333:4444::/64"'  || fail
+${CSCLI} decisions list -i aaaa:2222:3333:4445:5555:6666:7777:8888 -o json | jq -e '. == null' || fail
+${CSCLI} decisions list -i aaa1:2222:3333:4444:5555:6666:7777:8887 -o json | jq -e '. == null' || fail
+#check subrange within/out of range
+${CSCLI} decisions list -r aaaa:2222:3333:4444:5555::/80 -o json | jq -e '.[].decisions[0].value == "aaaa:2222:3333:4444::/64"'  || fail
+${CSCLI} decisions list -r aaaa:2222:3333:4441:5555::/80 -o json | jq -e '. == null'  || fail
+${CSCLI} decisions list -r aaa1:2222:3333:4444:5555::/80 -o json | jq -e '. == null'  || fail
+#check outer range
+${CSCLI} decisions list -r aaaa:2222:3333:4444:5555:6666:7777:8888/48 -o json | jq -e '. == null' || fail
+${CSCLI} decisions list -r aaaa:2222:3333:4444:5555:6666:7777:8888/48 -o json --contained | jq -e '.[].decisions[0].value == "aaaa:2222:3333:4444::/64"' || fail
+
```

Comment:
may be this one aaa:2222:3333:4445:5555:6666:7777:8888/48 -o json --contained | jq -e '. == null' || fail

Reply:
done